examples/job_api.rs

```diff
@@ -73,13 +73,11 @@ async fn main() -> anyhow::Result<()> {
     info!("Deleting the job record.");
     jobs.delete("empty-job", &DeleteParams {
         dry_run: true,
-        ..DeleteParams::default()
+        propagation_policy: Some(PropagationPolicy::Background),
+        ..Default::default()
```

Comment:
Should this be changed back to `..DeleteParams::default()`? Or is using the standard fault always the same?

Reply:
For dry-run, it doesn't matter, functionally. But we should limit the confusion either way;

We should have builders for the remaining portions in DeleteParams as well so that this can become DeleteParams::background().dry_run()

Reply:
Should all the properties of DeleteParams be private in that case? Why not have one `new()` constructor in that case.

Reply:
Looking at how we handle this for the other param types, no, the fields should probably remain public.

Reply:
I added setters for the remaining fields of delete params and modified the example to use `dry_run()`. It seems you can set all fields with the setters and they could then be private but I'm happy with pushing forward as is.